File: src/gestorAplicacion/serviciosExtra/ServiciosExtra.java
package gestorAplicacion.serviciosExtra;

public enum ServiciosExtra {
    GIMNASIO(1, 50000),
    BAR(2, 20000),
    PARQUEADERO(3, 30000);

    private final int idServicio;
    private final int precioServicio;

    ServiciosExtra(final int idServicio, final int precioServicio) {
        this.idServicio = idServicio;
        this.precioServicio = precioServicio;
    }

    public int getPrecio() {
        return precioServicio;
    }

    public int getIdServicio() {
        return idServicio;
    }

    public static ServiciosExtra buscarPorId(final int idServicio) {
        for (ServiciosExtra servicio : ServiciosExtra.values()) {
            if (servicio.getIdServicio() == idServicio) {
                return servicio;
            }
        }
        return null;
    }
    
    // Método para actualizar precios
    //! No sirve si la variables es final
    // public void actualizarPrecio(final int nuevoPrecio) {
    //     this.precioServicio = nuevoPrecio;
    // }
    
    //

    // Método para agregar nuevos servicios
    // public static void agregarServicioExtra(final String nombre, final int precio) {
    //     ServiciosExtra[] nuevosServicios = new ServiciosExtra[ServiciosExtra.values().length + 1];
    //     System.arraycopy(ServiciosExtra.values(), 0, nuevosServicios, 0, ServiciosExtra.values().length);
    //     nuevosServicios[nuevosServicios.length - 1] = new ServiciosExtra(nuevosServicios.length, precio);
    //     ServiciosExtra.values(nuevosServicios);
    // }
    
    public static int calcularPrecioServiciosExtras(int[] idsServicios) {
        int precioTotal = 0;
        for (int idServicio : idsServicios) {
            ServiciosExtra servicio = ServiciosExtra.buscarPorId(idServicio);
            if (servicio != null) {
                precioTotal += servicio.getPrecio();
            }
        }
        return precioTotal;
    }
    
}

File: python/src/gestor_aplicacion/servicios_extra/Servicios_extra.py
from enum import Enum

class ServiciosExtra(Enum):
    
    GIMNASIO = (1, 50000)
    BAR = (2, 20000)
    PARQUEADERO = (3, 30000)
    EXCURSION = (4, 150000)
    CHIVA = (5, 80000)


    def __init__(self, idServicio, precioServicio):
        self.idServicio = idServicio
        self.precioServicio = precioServicio


    def getPrecio(self):
        return self.precioServicio


    def getIdServicio(self):
        return self.idServicio


    @staticmethod
    def buscarPorId(idServicio):
        for servicio in ServiciosExtra:
            if servicio.getIdServicio() == idServicio:
                return servicio
        return None


    @staticmethod
    def calcularPrecioServiciosExtras(idsServicios):
        precioTotal = 0
        for idServicio in idsServicios:
            servicio = ServiciosExtra.buscarPorId(idServicio)
            if servicio is not None:
                precioTotal += servicio.getPrecio()
        return precioTotal

